core/src/program.rs

```diff
@@ -17,16 +16,41 @@ use super::Kernel;
 
 #[derive(Clone, Debug)]
 pub struct Program {
-    mast_forest: MastForest,
+    mast_forest: Arc<MastForest>,
```

Comment:
This was a result of executing a `Program` instead of `MastForest` in the `execute_*()` processor methods. Fixed now!

Reply:
@bobbinth I'd actually consider using `Rc` or `Arc` for forests, so that we can clone forests cheaply. If our intent is to keep them read-only once compiled (and/or loaded from a package), then storing them in a reference-counted structure makes a ton of sense. Otherwise we end up having to deep clone forests from the object store into a given `Program`

Reply:
I'm okay with either direction. AFAICT we currently don't need any deep clone, but these might become necessary in future PRs? Maybe we add an `Arc` if they become necessary then?

Reply:
Slight preference to keep them as is until we have a need for deep clone - but I'm also fine either way.